### DISCUSSION TITLE
Event on the third saturday of every month

### DISCUSSION THREAD

**Discussion by lewisevans777** [add2cal/add-to-calendar-button#566] [Q&A]
Is it possible to set the add to calendar to specify an event that happens on the third saturday of every month? if so, how?

**Comment by jekuer** [comment]
Yes!

You would either set:
```
recurrence="monthly"
recurrence_interval="1"
recurrence_byDay="3SA"
```

or use an RRULE like this:
```
recurrence="RRULE:FREQ=MONTHLY;BYDAY=3SA"
```

Mind that the date you set needs to be a third Saturday as well. Otherwise, it breaks at the calendar provider.

**Comment by lewisevans777** [comment]
Excellent! Many thanks. I would never have worked that out!

 

Lewis

 

From: Jens Kuerschner ***@***.***>
Reply to: add2cal/add-to-calendar-button ***@***.***>
Date: Wednesday 13 March 2024 at 17:47
To: add2cal/add-to-calendar-button ***@***.***>
Cc: Lewis Evans ***@***.***>, Author ***@***.***>
Subject: Re: [add2cal/add-to-calendar-button] Event on the third saturday of every month (Discussion #566)

 

Yes!

You would either set:
recurrence="monthly"
recurrence_interval="1"
recurrence_byDay="3SA"
or use an RRULE like this:
recurrence="RRULE:FREQ=MONTHLY;BYDAY=3SA"
Mind that the date you set needs to be a third Saturday as well. Otherwise, it breaks at the calendar provider.

**Reply by lewisevans777** [reply]
Well, I have followed the instructions and added a couple of other things, but it won't show up. Can you tell me what is wrong in this code?

<add-to-calendar-button name="Monthly Pub World Gathering" startdate="2024-03-16" starttime="11:30”
  endTime=“12:30" timezone="America/Los_Angeles" location="" description="To join, make sure you are registered on the website at:[br]→[url]https://thewhysmen.com/monthly-pub-world-gathering[/url][br] to get access to the Zoom link." options="'Apple','Google','iCal','Outlook.com','Microsoft 365','Microsoft Teams','Yahoo'" recurrence="RRULE:FREQ=MONTHLY;INTERVAL=1;BYDAY=3SA" lightmode="bodyScheme" atcb-button-id="atcb-btn-28" style="visibility: visible; opacity: 1; position: relative;"></add-to-calendar-button>

Many thanks.

**Reply by lewisevans777** [reply]
Oh, the code didn't copy.

 `<add-to-calendar-button
  name="Monthly Pub World Gathering"
  startDate="2024-03-16"
  startTime="11:30”
  endTime=“12:30"
  timeZone="America/Los_Angeles"
  location=""
  description="To join this meeting, make sure you are registered on the website at :[br]→[url]https://thewhysmen.com/monthly-pub-world-gathering[/url] to get access to the Zoom link."
  options="'Apple','Google','iCal','Microsoft 365','Microsoft Teams','Outlook.com','Yahoo'"
  recurrence="RRULE:FREQ=MONTHLY;BYDAY=3SA"
  lightMode="bodyScheme"
></add-to-calendar-button>`

**Reply by jekuer** [reply]
Can you add the debug option (simply add "debug" as attribute) and share what the browser console says?

**Comment by lewisevans777** [comment]
I’m sorry, I am not a developer. I don’t understand this instruction. Add debug where? How do I add it as an attribute? 

 

Forgive me. This is foreign territory.

 

Lewis

 

From: Jens Kuerschner ***@***.***>
Reply to: add2cal/add-to-calendar-button ***@***.***>
Date: Wednesday 13 March 2024 at 21:13
To: add2cal/add-to-calendar-button ***@***.***>
Cc: Lewis Evans ***@***.***>, Author ***@***.***>
Subject: Re: [add2cal/add-to-calendar-button] Event on the third saturday of every month (Discussion #566)

 

Can you add the debug option (simply add "debug" as attribute) and share what the browser console says?

**Reply by jekuer** [reply]
no worries.
Like this:
`<add-to-calendar-button debug name="title" ... />`

**Comment by lewisevans777** [comment]
Ah! Magic. I get this:

 

 

From: Jens Kuerschner ***@***.***>
Reply to: add2cal/add-to-calendar-button ***@***.***>
Date: Wednesday 13 March 2024 at 21:55
To: add2cal/add-to-calendar-button ***@***.***>
Cc: Lewis Evans ***@***.***>, Author ***@***.***>
Subject: Re: [add2cal/add-to-calendar-button] Event on the third saturday of every month (Discussion #566)

 

no worries.
Like this:
<add-to-calendar-button debug name="title" ... />

**Reply by jekuer** [reply]
I guess images do not show up, when you reply by email.

**Comment by lewisevans777** [comment]
It says:

Error: Add to Calendar Button generation (atcb-btn-3) failed: time misspelled [-> HH:MM]

 

Lewis

 

From: Jens Kuerschner ***@***.***>
Reply to: add2cal/add-to-calendar-button ***@***.***>
Date: Thursday 14 March 2024 at 09:20
To: add2cal/add-to-calendar-button ***@***.***>
Cc: Lewis Evans ***@***.***>, Author ***@***.***>
Subject: Re: [add2cal/add-to-calendar-button] Event on the third saturday of every month (Discussion #566)

 

I guess images do not show up, when you reply by email.

**Comment by jekuer** [comment]
Then that's the issue 😃 .
I am 98% sure that the two quotation marks in my screenshot above are no clean double quotes, but some special character:
<img width="170" alt="image" src="https://github.com/add2cal/add-to-calendar-button/assets/8572883/fe3b57d9-7fe4-45f2-ad10-84a7fdeb0e42">

Usually, this happens, when preparing this code in Word or similar text editors.

**Comment by lewisevans777** [comment]
Yes! That worked. 😊

 

The only bit that doesn’t work now are the icons for 'Outlook.com','Microsoft 365','Microsoft Teams','Yahoo'.

 

Any ideas? I have used them before but they don’t show this time.

 

Many thanks for your help!

 

Lewis

 

From: Jens Kuerschner ***@***.***>
Reply to: add2cal/add-to-calendar-button ***@***.***>
Date: Thursday 14 March 2024 at 20:19
To: add2cal/add-to-calendar-button ***@***.***>
Cc: Lewis Evans ***@***.***>, Author ***@***.***>
Subject: Re: [add2cal/add-to-calendar-button] Event on the third saturday of every month (Discussion #566)

 

Then that's the issue 😃 .
I am 98% sure that the two quotation marks in my screenshot above are no clean double quotes, but some special character:
image.png (view on web)

Usually, this happens, when preparing this code in Word or similar text editors.

**Reply by jekuer** [reply]
Not without looking at the actual implementation.

**Reply by lewisevans777** [reply]
It's the same code as above in this thread, and I have checked the quotation marks are the same as the ones that worked before.
`
options="'Apple','Google','iCal','Microsoft 365','Microsoft Teams','Outlook.com','Yahoo'"

`

**Reply by jekuer** [reply]
What I meant: The code seems correct. There is no obvious reason why the icons would not show up. So, this can have something to do with a lot of very specific things in your setup. From the script version to OS to device to other scripts on the same page.
I could maybe help when seeing it myself in production. Otherwise, it would be random guessing.

**Reply by lewisevans777** [reply]
OK, many thanks - yet again. How would I show it to you 'in production'?
Really appreciating you taking the time on this. I may not reply again for a while as I am driving....

**Reply by jekuer** [reply]
You can either send me a link to your website, where I can observe it - or you use something like CodePen to create a demo of the problem. Something like this.

If everything gets too complicated, you could also check out our PRO offering as it comes with way easier setup and management.

**Comment by lewisevans777** [comment]
Here's one where I thought it was working, although I now see several are not live links. https://lewisevans.net/cognac2023/
And here's the one I am working on now, that shows just three of them. https://thewhysmen.com/monthly-pub-world-gathering/

**Reply by jekuer** [reply]
Ah, got you.
When using recurrence, the button only shows the calendar options, which safely support this! That's why you are not seeing all of them.
I also noticed that your layout overlaps with the list. You can optimize your HTML/CSS or use the forceOverlay option.

Have a nice day!

**Comment by lewisevans777** [comment]
Many thanks, Jens – and for the tip on the overlay. I tink I can give the button a higher z value in Thrive.

 

Cheers,

 

Lewis

 

From: Jens Kuerschner ***@***.***>
Reply to: add2cal/add-to-calendar-button ***@***.***>
Date: Monday 18 March 2024 at 12:51
To: add2cal/add-to-calendar-button ***@***.***>
Cc: Lewis Evans ***@***.***>, Author ***@***.***>
Subject: Re: [add2cal/add-to-calendar-button] Event on the third saturday of every month (Discussion #566)

 

Ah, got you.
When using recurrence, the button only shows the calendar options, which safely support this! That's why you are not seeing all of them.
I also noticed that your layout overlaps with the list. You can optimize your HTML/CSS or use the forceOverlay option.

Have a nice day!
